docs/userguide/storagedriver/btrfs-driver.md

```diff
@@ -0,0 +1,266 @@
+<!--[metadata]>
++++
+title = "BTRFS storage in practice"
+description = "Learn how to optimize your use of BTRFS driver."
+keywords = ["container, storage, driver, BTRFS "]
+[menu.main]
+parent = "mn_storage_docker"
++++
+<![end-metadata]-->
+
+# Docker and BTRFS in practice
+
+Btrfs is a next generation copy-on-write filesystem that supports many advanced storage technologies that make it a good fit for Docker. Btrfs is included in the mainline Linux kernel and it's on-disk-format is now considered stable. However, many of its features are still under heavy development and users should consider it a fast-moving target.
+
+Docker's `btrfs` storage driver leverages many Btrfs features for image and container management. Among these features are thin provisioning, copy-on-write, and snapshotting.
+
+This article refers to Docker's Btrfs storage driver as `btrfs` and the overall Btrfs Filesystem as Btrfs.
+
+**Note**: The [Commercially Supported Docker Engine (CS-Engine)](https://www.docker.com/compatibility-maintenance) does not currently support the `btrfs` storage driver.
+
+## The future of Btrfs
+
+Btrfs has been long hailed as the future of Linux filesystems. With full support in the mainline Linux kernel, a stable on-disk-format, and active development with a focus on stability, this is now becoming more of a reality.
+
+As far as Docker on the Linux platform goes, many people see the `btrfs` storage driver as a potential long-term replacement for the `devicemapper` storage driver. However, at the time of writing, the `devicemapper` storage driver should be considered safer, more stable, and more *production ready*. You should only consider the `btrfs` driver for production deployments if you understand it well and have existing experience with Btrfs.
+
+## Image layering and sharing with Btrfs
+
+Docker leverages Btrfs *subvolumes* and *snapshots* for managing the on-disk components of image and container layers.  Btrfs subvolumes look and feel like a normal Unix filesystem. As such, they can have their own internal directory structure that hooks into the wider Unix filesystem.
+
+Subvolumes are natively copy-on-write and have space allocated to them on-demand
+from an underlying storage pool. They can also be nested and snapped. The
```

Comment:
And I'm British with a habit for using words that only I understand. In this instance though I'm confident this works ;-)